sip-test-kit/src/main/java/de/ikor/sip/foundation/testkit/workflow/whenphase/routeinvoker/impl/MailRouteInvoker.java

```diff
@@ -0,0 +1,82 @@
+package de.ikor.sip.foundation.testkit.workflow.whenphase.routeinvoker.impl;
+
+import static de.ikor.sip.foundation.testkit.util.TestKitHelper.resolveConsumer;
+
+import de.ikor.sip.foundation.testkit.workflow.whenphase.routeinvoker.RouteInvoker;
+import java.util.*;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.internet.MimeMessage;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.camel.CamelContext;
+import org.apache.camel.Endpoint;
+import org.apache.camel.Exchange;
+import org.apache.camel.component.mail.MailComponent;
+import org.apache.camel.component.mail.MailConsumer;
+import org.apache.camel.component.mail.MailEndpoint;
+import org.apache.camel.component.mail.MailMessage;
+import org.apache.camel.support.EmptyAsyncCallback;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.stereotype.Component;
+
+/** Invoker of exchange processing for {@link MailConsumer} */
+@Slf4j
+@Component
+@RequiredArgsConstructor
+@ConditionalOnClass(MailComponent.class)
+public class MailRouteInvoker implements RouteInvoker {
+
+  private final CamelContext camelContext;
+
+  @Override
+  public Optional<Exchange> invoke(Exchange exchange) {
+    MailConsumer mailConsumer = (MailConsumer) resolveConsumer(exchange, camelContext);
+
+    Exchange mailExchange = mailConsumer.createExchange(true);
+    mailExchange.setMessage(createMailMessage(exchange));
+
+    mailConsumer.getAsyncProcessor().process(mailExchange, EmptyAsyncCallback.get());
+    return Optional.empty();
+  }
+
+  @Override
+  public boolean isApplicable(Endpoint endpoint) {
+    return endpoint instanceof MailEndpoint;
+  }
+
+  private MailMessage createMailMessage(Exchange exchange) {
```

Comment:
Code clearness could benefit if we name Exchange parameter a bit more descriptive.

We use Exchanges for two purposes in test kit: 1. to define test expectations and 2. as an payload for test execution and we should distinguish the two with descriptive naming. 
This applies to the whole class (well module in general) but on this method its easy to comment.
Here Exchange